### PR TITLE
feat(Print Order): track delivery and invoice status

### PR DESCRIPTION
**Pull request by haiderk0987**

### CODE REVIEW

File: digital_printing/digital_printing/doctype/print_order/print_order.py [hunk 1]
```diff
@@ -588,44 +707,78 @@ def create_work_orders(print_order):
...
+	delivery_notes = _get_delivery_notes_to_be_billed(filters=delivery_note_filters)
+
+	if not delivery_notes:
+		frappe.throw(_("There are no Delivery Notes to be billed"))
+
+	for d in delivery_notes:
+		target_doc = make_sales_invoice(d.name, target_doc=target_doc)
+
+	# Remove Taxes (so they are reloaded)
+	target_doc.taxes_and_charges = None
```
Comment (SaiFi0102) [inline digital_printing/digital_printing/doctype/print_order/print_order.py]: remove this block

File: digital_printing/digital_printing/doctype/print_order/print_order.js [hunk 2]
```diff
@@ -282,17 +314,32 @@ erpnext.digital_printing.PrintOrder = class PrintOrder extends frappe.ui.form.Co
...
+			method: "digital_printing.digital_printing.doctype.print_order.print_order.get_sales_invoice",
 			args: {
-				customer: this.frm.doc.customer
+				"print_order": this.frm.doc.name,
 			},
-			callback: (r) => {
-				if (r.message) {
-					this.frm.set_value(r.message);
+			callback: function (r) {
+				console.log(r)
```
Comment (SaiFi0102) [inline digital_printing/digital_printing/doctype/print_order/print_order.js]: Remove `console.log`

File: digital_printing/digital_printing/doctype/print_order/print_order.py [hunk 3]
```diff
@@ -588,44 +707,78 @@ def create_work_orders(print_order):
...
 
-	frappe.msgprint(_("Work Orders Created: {0}").format(
-		', '.join([frappe.utils.get_link_to_form('Work Order', wo) for wo in wo_list])
-	), indicator='green')
+	if wo_list:
+		frappe.msgprint(_("Work Orders Created: {0}").format(
+			', '.join([frappe.utils.get_link_to_form('Work Order', wo) for wo in wo_list])
+		), indicator='green')
+	else:
+		frappe.msgprint(_("Work Order already created in Draft."))
```
Comment (SaiFi0102) [inline digital_printing/digital_printing/doctype/print_order/print_order.py:715]: Are you sure?

File: digital_printing/digital_printing/doctype/print_order/print_order.py [hunk 4]
```diff
@@ -588,44 +707,78 @@ def create_work_orders(print_order):
...
-		fn_map.qty_type_fn = 'default_qty_type'
+	for d in sales_orders:
+		target_doc = make_delivery_note(d.name, target_doc=target_doc)
 
-	elif doc.doctype == "Print Order Item":
-		fn_map.uom_fn = 'uom'
-		fn_map.length_uom_fn = 'length_uom'
-		fn_map.qty_type_fn = 'qty_type'
+	# Remove Taxes (so they are reloaded)
+	target_doc.taxes_and_charges = None
```
Comment (SaiFi0102) [inline digital_printing/digital_printing/doctype/print_order/print_order.py]: remove this block

**Review by SaiFi0102 — CHANGES_REQUESTED** [review]

File: digital_printing/digital_printing/doctype/print_order/print_order.py (hunk 3 above)
Comment (haiderk0987) [reply digital_printing/digital_printing/doctype/print_order/print_order.py:715]: pretty much...

**Review by SaiFi0102 — APPROVED** [review]

**Pull request merged**